### PR TITLE
chore: Adding registration validations OIDC Share

### LINKED ISSUE #1558: [oidcvp] relying party registration
Implement auth request validations defined in https://openid.net/specs/openid-connect-self-issued-v2-1_0.html#section-9

### LINKED ISSUE #1628: [oidc4vp] relying party registration (Non-Pre registered Relying Party)

### PR DESCRIPTION
**Pull request by victro**
* Created an endpoint for registration metadata
* Created an endpoint for jwk public keys
* Added signature to id token
Closes #1628 
Closes #1558

### CODE REVIEW

**Review by sudeshrshetty — CHANGES_REQUESTED** [review]
Do not merge

File: cmd/wallet-web/src/pages/OIDCShare.vue [hunk 1]
```diff
@@ -295,6 +294,27 @@ export default {
...
+        // data from jwks_uri
+        // https://datatracker.ietf.org/doc/html/rfc7517s
+        const jwksUri = registrationMetadata.jwks_uri;
+        const jwksResponse = await fetch(jwksUri);
+        const jwksKeys = await jwksResponse.json();
+        // We will get an array of keys
+        // we may need to search for the specific public keys in the future
+        // taking the first one for now
+        this.rpPublicKey = jwksKeys.keys[0];
+        this.generateIdToken(this.rpPublicKey);
```
Comment (rolsonquadras) [inline cmd/wallet-web/src/pages/OIDCShare.vue:314]: This doesn't look correct - we should use rp keys to validate the request (not signing the id_token at the provider end).

File: cmd/wallet-web/src/pages/OIDCShare.vue [hunk 2]
```diff
@@ -295,6 +294,27 @@ export default {
...
+        this.generateVPToken();
```
Comment (rolsonquadras) [inline cmd/wallet-web/src/pages/OIDCShare.vue:315]: Can we move generate functions outside of validate func ? Lets have single responsibility functions.

Comment (rolsonquadras) [inline cmd/wallet-web/src/pages/OIDCShare.vue:315]: Can we move generate functions outside of validate func ? Lets have single responsibility functions.

File: test/mock/adapter/adapter.go [hunk 3]
```diff
@@ -150,6 +150,12 @@ func startAdapterApp(agent *didComm, router *mux.Router) error {
 	router.HandleFunc("/verifier/oidc", app.oidcVerifier)
 	router.HandleFunc("/verifier/oidc/share", app.oidcShare)
 	router.HandleFunc("/verifier/oidc/share/cb", app.oidcShareCallback)
+	// Registration metadata endpoint that contains data according to the specification:
+	// https://openid.net/specs/openid-connect-registration-1_0.html#ReadResponse
+	router.HandleFunc("/.well-known/openid-configuration", app.oidcVerifierRegistrationMetadata).Methods(http.MethodGet)
```
Comment (rolsonquadras) [inline test/mock/adapter/adapter.go:155]: This looks to be OpenIDConnect config file and not registration url.

File: test/mock/adapter/adapter.go [hunk 4]
```diff
@@ -352,6 +358,9 @@ func (v *adapterApp) oidcShare(w http.ResponseWriter, r *http.Request) {
 	q := req.URL.Query()
 	q.Add("client_id", "demo-verifier")
 	q.Add("redirect_uri", os.Getenv(demoExternalURLEnvKey)+"/verifier/oidc/share/cb")
+	// Adding registration_uri param according to spec:
+	// https://openid.net/specs/openid-connect-self-issued-v2-1_0.html#name-relying-party-registration-
+	q.Add("registration_uri", os.Getenv(demoExternalURLEnvKey) + "/.well-known/openid-configuration")
```
Comment (rolsonquadras) [inline test/mock/adapter/adapter.go:363]: This looks to be OpenIDConnect config file and not registration url.

Comment (rolsonquadras) [inline test/mock/adapter/adapter.go:363]: This looks to be OpenIDConnect config file and not registration url.

**Pull request closed without merging**